src/Npgsql/Internal/TypeMapping/TypeMapper.cs

```diff
@@ -265,8 +284,7 @@ internal NpgsqlTypeHandler ResolveByDataTypeName(string typeName)
     {
         lock (_writeLock)
         {
-            if (DatabaseInfo.GetPostgresTypeByName(typeName) is not { } pgType)
-                throw new NotSupportedException("Could not find PostgreSQL type " + typeName);
+            var pgType = DatabaseInfo.GetPostgresTypeByName(typeName);
```

Comment:
I think this removes the distinction between a type not found in the database and a programming error where we added a type of type (like multirange) and forgot to handle it in the switch... Maybe add back null in the switch with the previous error message?

Reply:
`GetPostgresTypeByName` already throws an exception if the type is not found, we have a different method (`TryGetPostgresTypeByName`) specifically for that. That was the main reason for this change.

Reply:
Ah I see, right - it even returns non-nullable. I wonder why the IDE doesn't warn about the previous null check then.